In [3]:
import pandas as pd

In [40]:
df_rainbow = pd.read_csv('bc_rainbow_list.tsv', sep = '\t')

In [41]:
df_gbif = pd.read_csv('GBif Total.csv', sep = '\t', dtype={"gbifID": int})

In [42]:
df_gbif = df_gbif[['species', 'genus']]
df_gbif = df_gbif.drop_duplicates(subset=['species'])
df_gbif = df_gbif.reset_index()
df_gbif = df_gbif.drop(['index'], 1)
df_gbif['species'] = df_gbif['species'].astype(str)
df_gbif['specific'] = df_gbif['species'].apply(lambda x: x.split(" ")[-1])

In [24]:
df_gbif.iloc[]

species     Neriene digna
genus             Neriene
specific            digna
Name: 0, dtype: object

In [43]:
gp = df_gbif.groupby('specific')
indexes_gbif = gp.size().index.to_list()
values_gbif = list(gp.size().values)

In [29]:
values_gbif

[1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 6,
 2,
 1,
 2,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 5,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 2,
 3,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 2,
 1,
 1,
 3,
 2,
 7,
 7,
 2,
 4,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 1,
 4,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 2,
 1,
 1,
 1,
 1,
 3,
 2,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 3,
 2,
 1,
 1,
 1,
 1,
 3,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 3,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 5,
 18,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 8,
 1,
 1,
 1,
 2,
 19,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 6,
 4,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 3,
 5,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 3,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1

In [44]:
df_rainbow = df_rainbow[['Scientific Name']]
df_rainbow['species'] = df_rainbow['Scientific Name']
df_rainbow = df_rainbow.drop(['Scientific Name'], 1)
df_rainbow['specific'] = df_rainbow['species'].apply(lambda x: x.split(" ")[-1])
df_rainbow['genus'] = df_rainbow['species'].apply(lambda x: x.split(" ")[0])

In [68]:
gp = df_rainbow.groupby('specific')
indexes_rain = gp.size().index.to_list()
values_rain = list(gp.size().values)

In [59]:
list(gp.size().values)

[15,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 7,
 1,
 1,
 3,
 3,
 3,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 4,
 1,
 1,
 1,
 2,
 1,
 1,
 2,
 1,
 1,
 2,
 2,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 2,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 3,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 5,
 2,
 4,
 4,
 1,
 1,
 1,
 1,
 3,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 2,
 1,
 1,
 1,
 3,
 1,
 1,
 2,
 1,
 1,
 2,
 1,
 2,
 1,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 3,
 1,
 4,
 1,
 4,
 1,
 1,
 1,
 1,
 1,
 1,
 8,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 6,
 15,
 1,
 1,
 1,
 2,
 2,
 1,
 1,
 2,
 2,
 4,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 2,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 1,
 6,
 2,
 1,
 1,
 1,
 2,
 4,
 2,
 6,
 1,
 1,
 1,
 1,
 4,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 8,
 4,
 1,
 2,
 1,
 1,
 1,
 2,
 1,
 1,
 5,
 1,
 2,
 1,
 2,
 1,
 1,
 1

In [77]:
nice = 0
bad = 0
worst = 0
species_not_found = 0
for gbif_ind, gbif_val in zip(indexes_gbif, values_gbif):
    if (gbif_ind in indexes_rain):
        list_ind_rain = indexes_rain.index(gbif_ind)
        rain_val = values_rain[list_ind_rain]
        if ( rain_val == gbif_val):
            nice = nice + 1
        elif (gbif_val > rain_val):
            worst = worst + 1
        else:
            bad = bad + 1
    else:
        species_not_found = species_not_found + 1

In [65]:
list_ind_rain = indexes_rain.index(indexes_gbif[1])

In [94]:
df_rainbow.mode(1)

0                                       1               2
0        Abagrotis                      Abagrotis apposita        apposita
1        Abagrotis                        Abagrotis baueri          baueri
2        Abagrotis                 Abagrotis brunneipennis   brunneipennis
3        Abagrotis                        Abagrotis cupida          cupida
4        Abagrotis                       Abagrotis dickeli         dickeli
5        Abagrotis                   Abagrotis discoidalis     discoidalis
6        Abagrotis                          Abagrotis dodi            dodi
7        Abagrotis                        Abagrotis duanca          duanca
8        Abagrotis                      Abagrotis erratica        erratica
9        Abagrotis                       Abagrotis forbesi         forbesi
10       Abagrotis                        Abagrotis glenni          glenni
11       Abagrotis                       Abagrotis hermina         hermina
12       Abagrotis                     Abagrotis mirabilis       mirabilis
13       Abagrotis                       Abagrotis nanalis         nanalis
14       Abagrotis                      Abagrotis nefascia        nefascia
15       Abagrotis                         Abagrotis orbis           orbis
16       Abagrotis                       Abagrotis placida         placida
17       Abagrotis                     Abagrotis pulchrata       pulchrata
18       Abagrotis                         Abagrotis reedi           reedi
19       Abagrotis                      Abagrotis scopeops        scopeops
20       Abagrotis                       Abagrotis trigona         trigona
21       Abagrotis                    Abagrotis turbulenta      turbulenta
22       Abagrotis                       Abagrotis variata         variata
23       Abagrotis                    Abagrotis vittifrons      vittifrons
24          Abdera                            Abdera firma           firma
25           Abies                          Abies amabilis        amabilis
26           Abies                           Abies grandis         grandis
27           Abies        Abies lasiocarpa var. lasiocarpa      lasiocarpa
28     Abietinella                    Abietinella abietina        abietina
29         Abronia                       Abronia latifolia       latifolia
...            ...                                     ...             ...
14156   Zeugophora                  Zeugophora scutellaris     scutellaris
14157   Zeugophora                      Zeugophora varians         varians
14158       Zilora                          Zilora hispida         hispida
14159       Zilora                     Zilora occidentalis    occidentalis
14160      Ziphius                     Ziphius cavirostris     cavirostris
14161        Zizia                            Zizia aptera          aptera
14162      Zonitis                     Zonitis vermiculata     vermiculata
14163   Zonitoides                     Zonitoides arboreus        arboreus
14164   Zonitoides                      Zonitoides nitidus         nitidus
14165  Zonotrichia                  Zonotrichia albicollis      albicollis
14166  Zonotrichia                 Zonotrichia atricapilla     atricapilla
14167  Zonotrichia                  Zonotrichia leucophrys      leucophrys
14168  Zonotrichia                     Zonotrichia querula         querula
14169   Zoogenetes                        Zoogenetes harpa           harpa
14170     Zophodia                 Zophodia grossulariella  grossulariella
14171         Zora                            Zora hespera         hespera
14172     Zornella                         Zornella armata          armata
14173     Zornella                      Zornella cryptodon       cryptodon
14174    Zorochros                      Zorochros caurinus        caurinus
14175    Zorochros                     Zorochros dispersus       dispersus
14176      Zostera                          Zostera marina          marina
14177  Zosteropoda                    Zost

In [78]:
nice

1924

In [79]:
bad

786

In [80]:
worst

694

In [81]:
species_not_found

4703

In [84]:
len(indexes_gbif)

8107

<h3> Bad scenario here. More than half of the specific species names in gbif do not occur in the yellow list </h3>

In [87]:
df_gbif = df_gbif[['genus']]
df_gbif = df_gbif.drop_duplicates(subset=['genus'])
df_gbif = df_gbif.reset_index()
df_gbif = df_gbif.drop(['index'], 1)

In [97]:
genus_gbif = list(df_gbif.genus.values)

In [112]:
df_rainbow = df_rainbow[['Scientific Name']]
df_rainbow['genus'] = df_rainbow['Scientific Name'].apply(lambda x: x.split(" ")[0])
df_rainbow = df_rainbow.drop(['Scientific Name'], 1)
df_rainbow = df_rainbow.drop_duplicates(subset=['genus'])
df_rainbow = df_rainbow.reset_index()
df_rainbow = df_rainbow.drop(['index'], 1)

In [114]:
genus_rain = list(df_rainbow.genus.values)

In [117]:
count = 0 
for gen_gbif in genus_gbif:
    if (gen_gbif in genus_rain):
        count = count + 1

In [118]:
count

1848

In [119]:
len(genus_gbif)

6642

In [120]:
len(genus_rain)

4256

<h3> Out of 6642 genera in gbif only 1848 exist in yellow list which has 4256 genera </h3>